### PR TITLE
refactor: Use custom Gomega matcher to assert VulnerabilityReports

### PR DESCRIPTION
**Pull request by danielpacak**
Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: itest/starboard/starboard_cli_test.go [hunk 1]
```diff
@@ -61,6 +64,63 @@ func NewPodSpec(podName string, containers map[string]string, args ...string) (*
...
+	return m.Match(actual)
+}
+
+func (matcher *vulnerabilityReportMatcher) FailureMessage(actual interface{}) (message string) {
+	return matcher.failureMessage
+}
+
+func (matcher *vulnerabilityReportMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	return matcher.negatedFailureMessage
+}
```
Comment (simar7) [inline itest/starboard/starboard_cli_test.go:122]: Just for my understanding: Where are the values for these fields initialized?

File: itest/starboard/starboard_cli_test.go [hunk 2]
```diff
@@ -61,6 +64,63 @@ func NewPodSpec(podName string, containers map[string]string, args ...string) (*
...
+func (matcher *vulnerabilityReportMatcher) Match(actual interface{}) (success bool, err error) {
+	_, ok := actual.(v1alpha1.VulnerabilityReport)
+	if !ok {
+		return false, fmt.Errorf("vulnerabilityReportMatcher expects a VulnerabilityReport")
+	}
+	gvk, err := apiutil.GVKForObject(matcher.owner, starboard.NewScheme())
+	if err != nil {
+		return false, err
+	}
+	m := MatchFields(IgnoreExtras, Fields{
```
Comment (simar7) [inline itest/starboard/starboard_cli_test.go:91]: For my understanding: What's MatchFields? Where is it defined?

Comment (danielpacak) [reply itest/starboard/starboard_cli_test.go:91]: With Ginkgo / Gomega a common approach is to use dot imports - see imports sections:

```go
import (
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	. "github.com/onsi/gomega/gbytes"
	. "github.com/onsi/gomega/gstruct"
)
```

This is syntactic sugar, you could also [avoid dot imports](https://onsi.github.io/ginkgo/#avoiding-dot-imports).

File: itest/starboard/starboard_cli_test.go (hunk 1 above)
Comment (danielpacak) [reply itest/starboard/starboard_cli_test.go:122]: A custom matcher is stateful and we set those values in the `Match(actual interface{})` callback. Lines 110-111.

Comment (danielpacak) [reply itest/starboard/starboard_cli_test.go]: However, I realised that I assigned messages before calling the `Match` method 🤦 I've already fixed that, but in general it is guaranteed that `FailureMessage` and `NegatedFailureMessage` will only be called after `Match`, so you can save off any state you need to compute the messages in `Match`. See https://onsi.github.io/gomega/#adding-your-own-matchers

File: itest/starboard/starboard_cli_test.go (hunk 2 above)
Comment (simar7) [reply itest/starboard/starboard_cli_test.go]: Ah lazy me should've opened the PR in an IDE to realize it was a dot import. That makes a lot more sense!

File: itest/starboard/starboard_cli_test.go (hunk 1 above)
Comment (simar7) [reply itest/starboard/starboard_cli_test.go]: I see that makes a lot more sense. Cool!

**Review by simar7 — APPROVED** [review]

**Pull request merged**